### PR TITLE
1145 email minor interest

### LINKED ISSUE #1145: Have an "Email All Interested" button on the CCE Minor Management page
Have an "Email All Interested" button to allow the admin to send a message to each student who has expressed interest in the CCE Minor.

Screenshot_20240313_115851

### PR DESCRIPTION
**Pull request by hoerstl**
## Issue
Have an "Email All Interested" button to allow the admin to send a message to each student who has expressed interest in the CCE Minor. Resolves #1145

## Solution
Make a button that emails everyone who's inside the interested students table. 
We did remove the ability to show how many people are in that table to put the button there (see image). 
### Before
image
### After
image
Let us know if this is a problem and we'll figure out something that keeps the functionality and is cosmetically awesome.

## To test
- Navigate to the minor management page under Admin in the sidebar.
image
- Click the email all button by the interested students table and ensure that a window opens with your system's default email application
- Ensure that the email addresses from the table match the users in the table.

### CODE REVIEW

**Comment by hoerstl** [comment]
[Have an "Email All Interested" button on the CCE Minor Management page](https://trello.com/c/WupWMMtP/735-have-an-email-all-interested-button-on-the-cce-minor-management-page)

**Review by azabeli — DISMISSED** [review]
This looks good to me and it works perfect. The only suggestion I would give is to maybe add some message as a sample email , if there is one.

**Comment by bledsoef** [comment]
I think it looks good overall. Can I ask why you removed the ability to filter how many people are shown in the table, though?

**Review by bledsoef — DISMISSED** [review]
Personally, I would not remove the page filtering in the top left and try to find a better place for the button. But if you guys feel that is the best way to do it I will approve it.

**Comment by andersoncedu** [comment]
> Personally, I would not remove the page filtering in the top left and try to find a better place for the button. But if you guys feel that is the best way to do it I will approve it.

This was going to be the same comment I made. Also, the mailto isn't working for me, it's not opening an email in outlook,

**Comment by hoerstl** [comment]
These are good concerns, I have moved the button. How do ya like it?
image

File: app/templates/admin/cceMinor.html [hunk 1]
```diff
@@ -45,21 +45,31 @@ <h2>CCE Minor Progress</h2>
...
 <table class="display" id="interestedStudentsTable">
     <thead>
         <tr>
             <th>Name</th>
         </tr>
     </thead>
     <tbody>
+        {% set interestedStudentEmails = [] %}
         {% for student in interestedStudentsList %}
+            {% set _ = interestedStudentEmails.append(student.email) %}
```
Comment (stevensonmichel) [inline app/templates/admin/cceMinor.html]: If there is a possibility to handle that in the backend before rendering, you should go for it. Having this operation done in the route would be awesome, given that jinja should be the last option for this kind of operation.

**Comment by stevensonmichel** [comment]
After clicking on Email All, the emails of the interested students are inproperly added, giving this impression of a whole long email.

[
Screenshot 2024-04-24 101231
](url)

**Comment by hoerstl** [comment]
> After clicking on Email All, the emails of the interested students are inproperly added, giving this impression of a whole long email.
> 
> [ Screenshot 2024-04-24 101231 ](url)

Thanks! This issue is fixed now.

File: app/templates/admin/cceMinor.html (hunk 1 above)
Comment (hoerstl) [reply app/templates/admin/cceMinor.html]: Good idea. This has been done

**Comment by andersoncedu** [comment]
You have tests failing for your PR.

**Review by AndersonStettner — COMMENTED** [review]
The only two things I have for now are the "X" on the flash message that I talked with Lawrence about and the `Email All` button. 

Due to how close the tables are the button, to me, looks like it could be for the CCE Minor Progress table or the Interested Students table. The suggestion I have for this is adding a bit of space between the two tables so they are a bit more separated. This is just a suggestions though, if yall are fine with it and others are then it is fine.

**Comment by hoerstl** [comment]
> The only two things I have for now are the "X" on the flash message that I talked with Lawrence about and the `Email All` button.
> 
> Due to how close the tables are the button, to me, looks like it could be for the CCE Minor Progress table or the Interested Students table. The suggestion I have for this is adding a bit of space between the two tables so they are a bit more separated. This is just a suggestions though, if yall are fine with it and others are then it is fine.

Comment addressed. I've both added a close button for sure onto the flashed messages and refactored how messages were being displayed from the backend to avoid redundant code. Additionally, I added some space between the two tables as per your suggestion which I liked.

**Review by AndersonStettner — APPROVED** [review]

**Pull request merged**